### ISSUE TITLE
Enable CUDA 13.0 binaries

### ISSUE THREAD

**Issue by tinglvv** [pytorch/pytorch#159779]
### 🚀 The feature, motivation and pitch

CUDA 13.0 is released this morning, creating issue tracker for CUDA 13.0 binaries enablement.

**Comment by malfet** [comment]
It would be nice to have a short description why updating to CUDA-13 is helpful. I.e. it might be that it improves perf, adds new features, new HW capabilities etc

**Comment by tinglvv** [comment]
> > Supporting new HW capabilities: sm_103 (B300/GB300), sm_110 (Jetson Thor), sm_120 (Geforce 5000 series), sm_121(DGX SPARK). Notes: sm_121 will be compatible with sm_120 binaries, sm_10x will be compatible with sm_100 binaries.
> 
> [@tinglvv](https://github.com/tinglvv) Aren't sm_120 already supported by CUDA-12.8 and rest is already supported by CUDA-12.9, isn't it? (Or can you clarify what HW were added as part of CUDA-12.9?)

Thanks for catching it, yes sm_120 is already supported in CUDA 12.8. sm_121 is the newly supported arch. Modified the description.

**Comment by nWEIdia** [comment]
https://developer.nvidia.com/blog/whats-new-and-important-in-cuda-toolkit-13-0/#wheel_package_changes_to_cuda_130 
it is also time to evaluate this.

**Comment by johnnynunez** [comment]
cudnn 9.12 is out @atalman @malfet

**Comment by eqy** [comment]
> > Hi! I managed to get it to build 2.9.0a0+gitc7609e1 with 13.0.48 Cuda13 over the weekend, took at least a dozen times compiling to work out all the errors, but the changes needed are actually pretty overall just really nitpicky. I haven't contributed before but I could post a PR or share a diff if it would help! Tested to confirm it works with some of the new CUDA13 features specifically just fine and is super fast on 5090-blackwell!
> 
> [@michaelw9999](https://github.com/michaelw9999) Congrats, do you have detailed notes? I'm having performance issues on 5090, `backward` is much slower than 4090, seems 5090 is not fully supported by 12.8/12.9, and cuda 13 explicitly mentioned supporting RTX 50 series. FYI, my test script and terminal output:
> 
> import torch, torch.nn as nn, torch.nn.functional as F
> import numpy as np
> import time
> 
> batch_size, seqlen, i_dim, hidden  = 2048,128,38,32
> if __name__ == '__main__':
>     print(torch.cuda.get_device_name())
>     print('torch:', torch.__version__)
>     torch.set_default_dtype(torch.float64)
>     net = nn.Sequential(nn.Linear(i_dim, hidden), nn.Tanh(), nn.Linear(hidden, 1)).to('cuda:0')
>     optim = torch.optim.Adam(net.parameters(), lr=1e-3)
>     X_b = torch.randn((batch_size,seqlen,i_dim), device='cuda:0')
>     y_true = torch.randn((batch_size,seqlen,1), device='cuda:0')
>     for epoch in range(4):
>         t_forward=0
>         t_backward=0
>         for _ in range(100):
>             optim.zero_grad(set_to_none=True)
>             t0=time.time()
>             pred = net(X_b)
>             loss = F.mse_loss(pred, y_true)
>             torch.cuda.synchronize()
>             t1=time.time()
>             loss.backward()
>             torch.cuda.synchronize()
>             t2=time.time()
>             t_forward+=(t1-t0)
>             t_backward+=(t2-t1)
>         print(f"{epoch}: forward: {t_forward:.2f}s, backward: {t_backward:.2f}s")
> NVIDIA GeForce RTX 4090
> torch: 2.9.0.dev20250807+cu128
> 0: forward: 0.21s, backward: 0.20s
> 1: forward: 0.14s, backward: 0.16s
> 2: forward: 0.14s, backward: 0.17s
> 3: forward: 0.14s, backward: 0.16s
> NVIDIA GeForce RTX 5090
> torch: 2.8.0+cu129
> 0: forward: 0.20s, backward: 5.40s
> 1: forward: 0.12s, backward: 5.36s
> 2: forward: 0.12s, backward: 5.37s
> 3: forward: 0.12s, backward: 5.34s

CUDA 13 is not required for full support of 5090. Please create a new issue for performance problems, as this topic is only for enabling the build(s).
